tests/System.Binary.Base64.Tests/BasicUnitTests.cs

```diff
@@ -186,6 +304,256 @@ public void DecodingOutputTooSmall()
         }
 
         [Fact]
+        public void SampleUsage1()
+        {
+            Span<byte> source = new byte[1000];
+            InitalizeDecodableBytes(source);
+
+            Span<byte> decodedBytes = new byte[Base64Encoder.ComputeDecodedLength(source)];
+
+            Assert.True(Base64Encoder.TryDecode(source, decodedBytes, out int consumed, out int written));
+            Assert.Equal(source.Length, consumed);
+            Assert.Equal(decodedBytes.Length, written);
+        }
+
+        [Fact]
+        public void SampleUsage2()
+        {
+            var rnd = new Random(42);
+            Span<byte> source = new byte[1000];
+            InitalizeDecodableBytes(source);
+
+            int numBytes = rnd.Next(100, 251);
+            Span<byte> decodedBytes = new byte[numBytes];
+
+            int bytesConsumed = 0;
+            int bytesWritten = 0;
+
+            bool success = false;
+            do
+            {
+                success = Base64Encoder.TryDecode(source.Slice(bytesConsumed), decodedBytes, out int consumed, out int written);
+                bytesConsumed += consumed;
+                bytesWritten += written;
+                numBytes = rnd.Next(100, 251);
+                decodedBytes = new byte[numBytes];
+            } while (!success);
+
+            Assert.Equal(source.Length, bytesConsumed);
+            Assert.Equal(Base64Encoder.ComputeDecodedLength(source), bytesWritten);
+        }
+
+        [Fact]
+        public void SampleUsage3()
```

Comment:
> Also, i don't think you need to unit tests all the stitching strategies. Just the one we currently think is the best one. The perf tests should validate after they run that the full data set was properly encoded/decoded.

Of course. I will add the validation within the perf test itself once we settle on what we want.

> One tests that implements the piping loop with InlineData for various combinations of buffers. Ideally the pipeline loop would be a reusable routine that can be called from both unit tests and performance tests.

Agreed. The SampleUsage tests will be removed and can be ignored for this PR/discussion.